src/Go/effects/effect.ts

```diff
@@ -0,0 +1,144 @@
+  const effectSize = formatPercent(CalculateEffect(nodePower, opponent));
+  const effectDescription = getEffectTypeForFaction(opponent);
+  return `${effectSize > 0 ? "+" : ""}${effectSize}% ${effectDescription}`;
+}
+
+/**
+ * Update the player object, using the multipliers gained from node power for each faction
+ */
+export function updateGoMults(): void {
+  const mults = calculateMults();
+  Player.mults = mergeMultipliers(Player.mults, mults);
+  Player.updateSkillLevels();
+}
+
+/**
+ * Creates a multiplier object based on the player's total node power for each faction
+ */
+function calculateMults(): Multipliers {
+  const mults = defaultMultipliers();
+  [
+    opponents.Netburners,
+    opponents.SlumSnakes,
+    opponents.TheBlackHand,
+    opponents.Daedalus,
+    opponents.Illuminati,
+  ].forEach((opponent) => {
+    if (!Player.go?.status?.[opponent]) {
+      Player.go = getGoPlayerStartingState();
+    }
+
+    const effect = CalculateEffect(Player.go.status[opponent].nodePower, opponent);
+    switch (opponent) {
+      case opponents.Netburners:
+        mults.hacknet_node_purchase_cost *= effect;
+        break;
+      case opponents.SlumSnakes:
+        mults.crime_success *= effect;
+        break;
+      case opponents.TheBlackHand:
+        mults.hacking_exp *= effect;
+        break;
+      case opponents.Daedalus:
+        mults.company_rep *= effect;
+        mults.faction_rep *= effect;
+        break;
+      case opponents.Illuminati:
+        mults.hacking_speed *= effect;
+        break;
+    }
+  });
+  return mults;
+}
+
+export function resetGoNodePower(player: PlayerObject) {
+  [
+    opponents.Netburners,
+    opponents.SlumSnakes,
+    opponents.TheBlackHand,
+    opponents.Daedalus,
+    opponents.Illuminati,
+  ].forEach((opponent) => {
+    player.go.status[opponent].nodePower = 0;
+    player.go.status[opponent].nodes = 0;
+    player.go.status[opponent].winStreak = 0;
+  });
+}
+
+export function playerHasDiscoveredGo() {
+  const playedGame = Player.go.boardState.history.length || Player.go.previousGameFinalBoardState?.history?.length;
+  const hasRecords = [
+    opponents.Netburners,
+    opponents.SlumSnakes,
+    opponents.TheBlackHand,
+    opponents.Daedalus,
+    opponents.Illuminati,
+  ].find((opponent) => Player.go.status[opponent].wins + Player.go.status[opponent].losses);
+  const isInBn14 = Player.bitNodeN === 14;
+
+  // TODO: remove this once testing is completed
+  const isInTesting = true;
+
+  return !!(playedGame || hasRecords || isInBn14 || isInTesting);
+}
+
+function formatPercent(n: number) {
+  return floor((n - 1) * 10000) / 100;
+}
+
+function getEffectPowerForFaction(opponent: opponents) {
+  return opponentDetails[opponent].bonusPower;
+}
+
+export function getEffectTypeForFaction(opponent: opponents) {
+  return opponentDetails[opponent].bonusDescription;
+}
+
+export function getWinstreakMultiplier(winStreak: number) {
+  return winStreak ? 1.2 ** (winStreak - 1) : 0.5;
```

Comment:
Since winStreak is exponential, a strategy that can guarantee successive wins will completely dominate any less-certain strategy. In the long run it will take log(2)/log(1.2^0.3) ~= 12.7 consecutive wins for each doubling of effect. (3.8 wins per doubling of nodePower.)
Maybe that's intended? But it just further cements the meta as "hook this up to KataGo," which leaves a bad taste in my mouth.

Reply:
I have changed this to cap winstreak at 3x bonus to prevent the exponential scaling mentioned..

I have also added a feature to give an extra bonus on the first win after a dry streak, going up to 5x bonus. This will reward to improvement and perseverance, but not get out of control. (Note that games must be completed, not just resigned, to count towards a losing streak for this purpose, to avoid cheese)